### ISSUE TITLE
Is scripts/firmwareupdater.ini file needed?

### ISSUE THREAD

**Issue by traversaro** [robotology/icub-firmware-build#177]
After launching some scripts as reported in https://github.com/robotology/icub-firmware-build/issues/176, I launched the FirmwareUpdater from the `scripts` directory, and I was quite surprised by the fact that it showed a lot of top level elements. I understood later that it was due to the fact that when launched in the `scripts` folder, the `FirmwareUpdater` was loading the file `firmwareupdater.ini` from https://github.com/robotology/icub-firmware-build/blob/master/scripts/firmwareupdater.ini, instead of the correct one from https://github.com/robotology/robots-configuration/blob/master/ergoCubSN002/firmwareupdater.ini . Perhaps to avoid this confusion we could remove the `scripts/firmwareupdater.ini` file?

@valegagge @MSECode

**Comment by traversaro** [comment]
For a similar reason, perhaps we could remove the https://github.com/robotology/icub-firmware-build/tree/6152a3d7325422c2f2be8a786e5f76700b9e80b4/scripts the `network.iCub***.xml` files present in scripts?

**Comment by MSECode** [comment]
> After launching some scripts as reported in [#176](https://github.com/robotology/icub-firmware-build/issues/176), I launched the FirmwareUpdater from the `scripts` directory, and I was quite surprised by the fact that it showed a lot of top level elements. I understood later that it was due to the fact that when launched in the `scripts` folder, the `FirmwareUpdater` was loading the file `firmwareupdater.ini` from https://github.com/robotology/icub-firmware-build/blob/master/scripts/firmwareupdater.ini, instead of the correct one from https://github.com/robotology/robots-configuration/blob/master/ergoCubSN002/firmwareupdater.ini . Perhaps to avoid this confusion we could remove the `scripts/firmwareupdater.ini` file?
> 
> [@valegagge](https://github.com/valegagge) [@MSECode](https://github.com/MSECode)

For me this might be fine.

**Comment by MSECode** [comment]
> For a similar reason, perhaps we could remove the https://github.com/robotology/icub-firmware-build/tree/6152a3d7325422c2f2be8a786e5f76700b9e80b4/scripts the `network.iCub***.xml` files present in scripts?

In this case, we should see, since that file might be used when updating automatically all the boards in the robot exploiting a bash script we developed some times ago (that actually need to be refined)

**Comment by traversaro** [comment]
> > After launching some scripts as reported in [#176](https://github.com/robotology/icub-firmware-build/issues/176), I launched the FirmwareUpdater from the `scripts` directory, and I was quite surprised by the fact that it showed a lot of top level elements. I understood later that it was due to the fact that when launched in the `scripts` folder, the `FirmwareUpdater` was loading the file `firmwareupdater.ini` from https://github.com/robotology/icub-firmware-build/blob/master/scripts/firmwareupdater.ini, instead of the correct one from https://github.com/robotology/robots-configuration/blob/master/ergoCubSN002/firmwareupdater.ini . Perhaps to avoid this confusion we could remove the `scripts/firmwareupdater.ini` file?
> > [@valegagge](https://github.com/valegagge) [@MSECode](https://github.com/MSECode)
> 
> For me this might be fine.

Great, I propose the removal in https://github.com/robotology/icub-firmware-build/pull/181 .

**Comment by valegagge** [comment]
Hi @traversaro, I don't understand if the `fileupdater.ini` file in the `script` folder caused the script not to work or if it just created confusion for the user.

I'd like to understand if the list of possible devices makes the update script faults. If we are in this case, for sure we need to remove this file from the folder script. Instead of removing it at all, we can keep it as a template. 

If it is just for confusion, maybe we can leave and update the [documentation](https://icub-tech-iit.github.io/documentation/icub_firmware/firmwareupdater/firmwareupdater/) accordingly.

What do you think?


cc @MSECode

**Comment by traversaro** [comment]
> Hi @traversaro, I don't understand if the fileupdater.ini file in the script folder caused the script not to work or if it just created confusion for the user.

No, the script worked "normally", with some errors but not related to the `firmwareupdater.ini` file. The problem is that if your current working directory is already the `scripts` folder (as you moved there to launch the scripts) and then you see some errors, the first thing that I did (as I was already there) was to launch the `FirmwareUpdater` GUI from the same directory. In that case the `FirmwareUpdater` loaded the `firmwareupdater.ini` contained in this repo, instead of the correct one contained in the robots-configuration repo. I understood what was going on as I am familiar with the fact that YARP modules first look for their configuration files in the current working directory, but this may not be obvious for non-expert users. If we really want to keep this file, can we at least move it in another directory, so that if you launch the `FirmwareUpdater` from the `scripts` folder it does not accidentally load the wrong `firmwareupdater.ini` file?

**Comment by valegagge** [comment]
Hi @traversaro ,
the file loaded is correct, but I agree with you that seeing the list of devices on which connect to search the embedded boards instead of the `ETH` one only could confuse the no-expert user.

So I think we have two options:
1. leave the file there and update the documentation accordingly
2. rename the file or move it in a different folder and keep it as a template file.

What do you think? Have you other ideas?

**Comment by traversaro** [comment]
Just to check if I understand correctly: that file is useful for expert users that use the Firmware update to connect directly to non-ETH boards?

**Comment by valegagge** [comment]
after a F2F discussion w/ @traversaro we concluded that the best solution is to remove the file since it is already in icub-main/src/tools/FirmwareUpdater as the default file. In addition ,it is mentioned in the [icub-tech Documentation.](https://icub-tech-iit.github.io/documentation/icub_firmware/firmwareupdater/firmwareupdater/)
